Feature/animfixed

Bathboomsoso opened the pull request:
기존 코드 반영

Bathboomsoso committed:
Feat: ALS 플러그인 통합 및 Shooter 캐릭터 상태/입력 확장

ALS 통합 단계에서 Shooter 캐릭터의 입력 시스템과 상태 처리를 강화함으로써 게임플레이 기능을 확장.

- `ShooterBaseCharacter`를 `AAlsCharacter`로 변경하고 ALS 관련 헤더 추가로 기능 통합.
- 무기 관련 입력(APIs 추가: 장착/해제, 발사 시작/중지, AIM, 리로드 등)을 바인딩하여 강화된 상호작용 제공.
- `UAlsCameraComponent` 추가 및 설정 전환(SwitchShoulder) 메서드 구현.
- `ShooterGamePlayTag`에 신규 태그 추가로 다중 입력 확장 지원.
- `Shooter.uproject`, `Shooter.Build.cs`에 ALS 및 ALSCamera 모듈의 의존성 반영.
- 기존 불필요한 코드 주석 처리 및 디버그 메시지 추가로 코드 효율 개선 및 유지보수 강화.

Bathboomsoso committed:
Feat: 기존 DEV 브랜치 인벤토리 시스템 초기 구현 및 입력 태그 추가

Shooter 캐릭터에 인벤토리 시스템 연동 및 신규 입력 태그 추가를 통해 게임플레이 기능을 확장.

- `InventoryComponent`를 Shooter 캐릭터에 추가하여 인벤토리 관리 기능 제공.
- `InputTag_OpenInventory`를 추가하여 인벤토리 열기 입력 동작 지원.
- `ShooterGamePlayTag.h/cpp`에 관련 태그 선언 및 정의 추가.
- 기존 입력 처리 메서드에 인벤토리 입력(`Input_OpenInventory`) 메서드를 바인딩.
- `GetShooterInventoryComponent` 메서드 추가로 인벤토리 접근성 개선.
- 코드 주석 및 불필요한 공백 정리로 코드 유지보수성 향상.

Bathboomsoso committed:
Fix : 기존 DEV 수정 사항 반영

Bathboomsoso committed:
Refactor: Shooter 캐릭터 입력 메서드 정리 및 가독성 개선

Shooter 캐릭터의 입력 메서드 순서를 재배치하고 명명 규칙을 수정하여 코드 가독성과 유지보수성을 향상시켰습니다.

- `Input_Reload` 메서드를 다른 입력 메서드와 함께 그룹화.
- `Input_Fire` 메서드에서 `Input_StartFire`로 이름 변경하여 명확한 의도 전달.
- 관련 메서드들의 정렬을 통해 코드 구조를 직관적으로 개선.

hwarang98 commented on Source/Shooter/Public/Characters/ShooterCharacter.h:
```diff
@@ -7,7 +7,7 @@
 #include "GameplayTagContainer.h"
 #include "ShooterCharacter.generated.h"
 
-class UCameraComponent;
```
요거 일단살려주실수있을까요? 제 캐릭터에서 카메라 쓰고있어서요!

hwarang98 commented on Source/Shooter/Public/Characters/ShooterCharacter.h:
```diff
@@ -44,14 +44,27 @@ class SHOOTER_API AShooterCharacter : public AShooterBaseCharacter
-private:
-#pragma region Components
-	UPROPERTY(VisibleAnywhere, BlueprintReadOnly, Category = "Camera", meta = (AllowPrivateAccess = "true"))
-	USpringArmComponent* CameraBoom;
```
요거 일단살려주실수있을까요? 제 캐릭터에서 카메라 쓰고있어서요!

Bathboomsoso committed:
Feat: ALS GameMode 변경 및 Shooter 캐릭터 무기 입력 수정

ALS GameMode를 기본 GameMode로 설정하고 Shooter 캐릭터의 무기 입력 방식을 개선하였습니다.

- `DefaultEngine.ini`의 `GlobalDefaultGameMode`를 ALS GameMode로 변경하여 ALS 기반 설정 적용.
- `Input_EquipWeapon`, `Input_UnequipWeapon` 메서드에서 인자를 제거하고 단순화된 호출 방식 구현.
- `ShooterCharacter.cpp`에서 불필요한 헤더(`#include`) 정리로 코드 경량화.
- `SetOverlayMode` 호출 시 추가 파라미터를 전달하여 더 명확한 동작 정의.

Bathboomsoso committed:
Feat: 구르기 신규 입력 추가및 무기 Overlay 버그수정

- `InputTag_Roll` 태그를 `ShooterGamePlayTag`에 선언 및 정의하여 구르기 입력 지원.
- `ShooterCharacter` 클래스에 `Input_Roll` 메서드 추가 및 `ShooterInputComponent`에 입력 바인딩.
- 구르기 동작 시 애니메이션 재생을 위해 `StartRolling` 호출과 속도 조정 적용.
- `DefaultEngine.ini` 파일에서 기본 및 에디터 시작 맵을 `NewMap`으로 변경.
- 불필요한 디버그 메시지를 포함한 코드 정리 및 주석 유지로 코드 일관성을 확보.

Bathboomsoso committed:
Feat: Shooter 캐릭터 카메라 컴포넌트 추가 [ 화랑님 요청 ]

Shooter 캐릭터에 기본 카메라 및 붐 암 컴포넌트를 추가하여 시점 설정과 확장성을 개선.

- `CameraBoom` 및 `FollowCamera` 멤버 변수를 추가하고 UPROPERTY 매크로를 통해 접근 가능하도록 설정.
- 기본 카메라 설정을 적용하여 Shooter 캐릭터의 시점 관리 기능 강화.
- 향후 카메라 관련 기능 확장을 대비한 구조 설계.

Bathboomsoso committed:
Refactor: ShooterCharacter.cpp 파일 내부 주석 추가 및 TODO 관리

코드 가독성과 유지보수성을 위해 주요 작업 상태를 지정하는 TODO 주석을 추가.

- 로직 간소화 및 설정 접근성 강화 목적으로 사용되지 않는 변수 및 함수들에 대한 제거 작업 예정.
- `RotationRate` 및 `MaxWalkSpeed` 등 불필요 변수의 ALS 호환성 및 설정 관리 간소화를 위한 계획 수립.
- 향후 리팩토링 방향성을 명확히 하기 위한 주석 작성.
